combat-trainer.lic

```diff
@@ -4830,13 +4833,53 @@ class GameState
     return unless can_engage?
     return if stomp
     return if pounce
+    return if rush
 
     case DRC.bput("engage", /^You (are already|begin to) (stealthily )?(advanc(e|ing)|at melee)/, /^There is nothing else/, /is already quite dead/, /^What do you want to advance towards/)
     when /You are already advancing/, /You begin to advance/, /You begin to stealthily advance/
       pause 2
     end
   end
 
+  def rush
+    return if retreating?
+    return false if DRC.left_hand
+    return false unless @rush_shield
+    return false unless npcs.any?
+    return false unless charged_maneuver_off_cooldown?(@charged_maneuvers['Shield Usage'])
+
+    wield_specific_weapon(@rush_shield, nil, false)
```

Comment:
For logging and future potential, I recommend specifying the skill `Shield Usage` when calling the wield weapon function. 

```suggestion
    wield_specific_weapon(@rush_shield, 'Shield Usage', false)
```

Despite the function name, it'll handle getting any gear to your hand. Today, the skill argument is for swappable weapons like ristes, but might as well pass this in since it won't hurt anything.